### PR TITLE
Fixing some edge cases in HLT-Patatrack customisations

### PR DESCRIPTION
**Pull request by missirol**
#### PR description:

This PR is an attempt to make the HLT-Patatrack customisations introduced in https://github.com/cms-sw/cmssw/pull/34956 a bit more robust (mostly to handle some edge cases, e.g. HLT configs with only a limited number of paths; see [this comment](https://github.com/cms-sw/cmssw/pull/34956#issuecomment-903461360) for an example).

Suggestions on the implementation are welcomed.

Attn: @fwyzard 

#### PR validation:

With the PR, the customisations are applied as intended in the following 3 tests (without PR, the first 2 tests wouldn't work correctly).

```bash
#!/bin/bash -ex

scram project CMSSW CMSSW_12_1_X_2021-08-23-2300
cd CMSSW_12_1_X_2021-08-23-2300/src
eval `scram runtime -sh`
git cms-merge-topic cmssw:34992
scram build

hltTest(){

  hltGetConfiguration /dev/CMSSW_12_0_0/GRun/V5 \
  --offline --mc --unprescale --process TEST \
  --input /store/relval/CMSSW_12_0_0_pre6/RelValTTbar_14TeV/GEN-SIM-DIGI-RAW/PU_120X_mcRun3_2021_realistic_v4-v1/00000/20a08d38-8b7f-4ce4-b24b-edc023ddd078.root \
  --output none \
  --era Run3 --globaltag 121X_mcRun3_2021_realistic_v1 \
  --customise HLTrigger/Configuration/customizeHLTforPatatrack.customizeHLTforPatatrackTriplets \
  --max-events 10 \
  "${@:2}" > "$1"_cfg.py

  edmConfigDump "$1"_cfg.py > "$1".py

  python3 "$1".py
  cmsRun "$1".py
}

hltTest hltPartial1 --paths HLTriggerFirstPath,MC_PFMET_v17,HLTriggerFinalPath
hltTest hltPartial2 --paths HLTriggerFirstPath,HLT_UncorrectedJetE30_NoBPTX_v6,HLTriggerFinalPath
hltTest hltFull
```

#### If this PR is a backport, please specify the original PR and why you need to backport that PR:

N/A

### CODE REVIEW

File: HLTrigger/Configuration/python/customizeHLTforPatatrack.py [hunk 1]
```diff
@@ -225,11 +224,10 @@ def customisePixelLocalReconstruction(process):
 # customisation for running the "Patatrack" pixel track reconstruction
 def customisePixelTrackReconstruction(process):
 
-    if not 'HLTRecoPixelTracksSequence' in process.__dict__:
+    if not all(seq in process.__dict__ for seq in ['HLTRecoPixelTracksSequence', 'HLTRecopixelvertexingSequence']):
```
Comment (fwyzard) [inline HLTrigger/Configuration/python/customizeHLTforPatatrack.py:227]: do we need/want to support the case where the configuration has `HLTRecoPixelTracksSequence` but not `HLTRecopixelvertexingSequence` ?

Comment (missirol) [reply HLTrigger/Configuration/python/customizeHLTforPatatrack.py]: I couldn't find a use case like this in the `GRun` menu, but supporting this seems easy enough.

Tentatively added in `e397a38176c`.

Checked that `e397a38176c` works as intended, by testing a config with
```diff
+process.MC_PixelTracksOnly_v0 = cms.Path(process.HLTRecoPixelTracksSequence)
+process.setSchedule_(cms.Schedule(process.MC_PixelTracksOnly_v0))
+process.prune()

#User-defined customization functions
from HLTrigger.Configuration.customizeHLTforPatatrack import customizeHLTforPatatrackTriplets
process = customizeHLTforPatatrackTriplets(process)
```

Comment (fwyzard) [reply HLTrigger/Configuration/python/customizeHLTforPatatrack.py]: Nice, thanks @missirol .

**Comment by Martin-Grunewald** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**